robolectric/build.gradle

```diff
@@ -31,18 +31,19 @@ dependencies {
     compile "org.bouncycastle:bcprov-jdk16:1.46"
     compile "com.ximpleware:vtd-xml:2.11"
     compile "com.thoughtworks.xstream:xstream:1.4.8"
-    provided "com.intellij:annotations:12.0"
+    compileOnly "com.intellij:annotations:12.0"
 
     compile "org.apache.ant:ant:1.8.0"
     compile("org.apache.maven:maven-ant-tasks:2.1.3") {
         exclude group: "junit", module: "junit"
     }
-    provided "org.robolectric:android-all:6.0.0_r1-robolectric-0"
-    provided "junit:junit:4.8.2"
+    compileOnly "org.robolectric:android-all:6.0.0_r1-robolectric-0"
+    compileOnly "junit:junit:4.8.2"
 
     // Testing dependencies
     testCompile "junit:junit:4.8.2"
     testCompile "org.hamcrest:hamcrest-core:1.3"
     testCompile "org.assertj:assertj-core:2.0.0"
     testCompile "org.mockito:mockito-core:1.8.0"
+    testRuntime "org.robolectric:android-all:6.0.0_r1-robolectric-0"
```

Comment:
Nice if we can get rid of that custom rule :-) do you also want to remove buildSrc/src/main/groovy/provided.groovy ?